pkg/nmi/server/server.go

```diff
@@ -74,7 +76,11 @@ func (s *Server) Run() error {
 	mux.Handle("/metadata/identity/oauth2/token/", appHandler(s.msiHandler))
 	mux.Handle("/host/token", appHandler(s.hostHandler))
 	mux.Handle("/host/token/", appHandler(s.hostHandler))
-	mux.Handle("/", appHandler(s.defaultPathHandler))
+	if s.BlockInstanceMetadata {
+		mux.Handle("/", http.NotFoundHandler())
```

Comment:
Can we please change this to StatusForbidden to reflect that we are blocking the calls ?

Reply:
I was torn between 403 and 404 here. In the end I picked 404 since it makes the code smaller since [the http package provides a handler for 404](https://golang.org/pkg/net/http/#NotFoundHandler). 404 is also sometimes used instead of 403 deliberately; for example, GitHub returns a 404 if you try to navigate to a private repo that you don't have access to.

Thoughts on the tradeoff between slightly more code and the slightly more correct HTTP code?

Reply:
Hah, turns out the NotFoundHandler implementation is trivial:

```go
// NotFoundHandler returns a simple request handler
// that replies to each request with a ``404 page not found'' reply.
func NotFoundHandler() Handler { return HandlerFunc(NotFound) }
```

So this can easily be changed to 403 😄